docs/src/operate/c8y/supported-operations.md

```diff
@@ -241,9 +243,177 @@ Arguments will be parsed correctly as long as following features are not include
+
+:::info
+Users can provide their own custom topic in the operation file:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command"
+[exec]
+topic = "${.bridge.topic_prefix}/custom/topic"
+on_fragment = "c8y_Command"
+command = "/etc/tedge/operations/command ${.payload.c8y_Command.text}"
+skip_status_update = true
+```
+
+Instead of providing `topic_prefix` manually, you can use `${.bridge.topic_prefix}` template to derive it from the bridge config of current cloud instance.
+
+The mapper will automatically subscribe to all custom topic provided in operation file under `topic` field when connecting to the cloud.  
+:::
+
+:::note
+When `on_fragment` is provided with `on_message`, the mapper will ignore the operation.
+:::
+
+For the second approach, the user is supposed to create a master custom operation definition file with `.template` extension, e.g.:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command.template"
+[exec]
+topic = "c8y/devicecontrol/notifications"
+on_fragment = "c8y_Command"
+
+[exec.workflow]
+operation = "command"
+input = "${.payload.c8y_Command.text}"
+```
+
+To indicate that devices are supporting the operation, the device must publish a command capability message where command name is derived from `exec.workflow.operation`:
+
+```sh
+tedge mqtt pub -r 'te/device/<name>///cmd/command' '{}'
+```
+
+On receiving, mapper creates a symlink at `/etc/tedge/operations/c8y/c8y_Command` (for main device) or `/etc/tedge/operations/c8y/<external_id>/c8y_Command` (for child device) to master template file.
+
+The Mapper detects the change in the `/etc/tedge/operations/c8y` directory except for `.template` file. When the change is detected, `114` (supported operation) message is sent to c8y.
+
+After that, C8Y Json message is converted to thin-edge command, preserving all parameters provided in `input` field.
+
+```text title="Topic"
+te/device/<name>///cmd/command/c8y-mapper-1800380
+```
+
+```json5 title="Payload"
+{
+      "status": "init",
+      "text":"echo helloworld",
+}
+```
+
+:::info
+An operation file definition can contain multiple `input` fields:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command.template"
+[exec]
+topic = "c8y/devicecontrol/notifications"
+on_fragment = "c8y_Command"
+
+[exec.workflow]
+operation = "command"
+input.x = "${.payload.c8y_Command.text}"
+input.y = { foo = "bar" }
+```
+
+Conversion to thin-edge command will contain both of them:
+
+```text title="Topic"
+te/device/<name>///cmd/command/c8y-mapper-1800380
+```
+
+```json5 title="Payload"
+{
+      "status": "init",
+      "text":"echo helloworld",
+      "foo":"bar",
+}
+```
+
+Make sure that provided input is json object, otherwise the operation execution will be skipped.
```

Comment:
Keep using the official names as much as possible.

```suggestion
Make sure that provided input is JSON object, otherwise the operation execution will be skipped.
```

Reply:
Resolved by a267f04581ec93b1644543df4d45e998b590aaa8